src/pkg/build/provisioners/provisioner_shub.go

```diff
@@ -9,23 +9,143 @@
 package provisioners
 
 import (
+	"encoding/json"
+        "errors"
 	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+//        "log"
+	"os"
+        "time"
+        "strings"
 
+	"github.com/containers/image/docker"
+	"github.com/containers/image/types"
+	oci "github.com/containers/image/oci/layout"
 	"github.com/singularityware/singularity/src/pkg/image"
+	"github.com/singularityware/singularity/src/pkg/sylog"
 )
 
-// ShubProvisioner provisions a sandbox environment from a shub
-// URI.
+
+// ShubClient contains the uri and client
+type ShubClient struct {
+	Client   http.Client
+	ImageUri string
+	HTTPAddr string
+}
+
+// NewRemoteBuilder creates a RemoteBuilder with the specified details.
+func NewShubClient(uri string) (shubClient *ShubClient) {
+
+	sylog.Infof("%v\n", uri)
+	httpAddr := fmt.Sprintf("www.singularity-hub.org")
```

Comment:
`fmt.Sprintf` probably unnecessary here. You can just change: 

`HTTPAddr: httpAddr,` 

on line 47 to:

`HTTPAddr: "www.singularity-hub.org",`

Go is, in general, much more lenient w.r.t. strings and memory allocation than C

Reply:
okay, I've adjusted this one step further, I put it directly into the data structure:

```
	sc = &shubClient{
		Client: http.Client{
			Timeout: 30 * time.Second,
		},
		HTTPAddr: "www.singularity-hub.org",
		ImageUri: uri,
	}

```